### PR TITLE
Framework: Pull in i18n-calypso

### PR DESCRIPTION
**Pull request by ockham**
As discussed for a while. WIP.

TODO:
- [x] Fix missing `interpolate-components` dependency. Probably going to `npm i` for now, and fold it into the repo in a subsequent step.
- [ ] Move back to `packages/`. Currently in `client/lib`. Webpack entry points can be a pain.
- [ ] Re-import i18n-calypso repo to preserve git history
- [ ] Clean up.

### CODE REVIEW

File: packages/i18n/package.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "i18n-calypso",
+  "version": "2.0.1",
+  "description": "i18n JavaScript library on top of Jed originally used in Calypso",
+  "main": "index.js",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/Automattic/i18n-calypso.git"
```
Comment (sirreal) [inline packages/i18n/package.json]: Flagging - this will need to be changed

**Comment by blowery** [comment]
also relevant here, maybe, #27356

**Comment by ockham** [comment]
Update: 

* Got it working, thanks to a debugging session with @sirreal (had to `npm i --save jed`
* Moved `i18n` back to `packages/`.

Current blocker: Y U NO TRANSPILE?

(Tried a bunch of things, like adding `packages/` to `resolve`, `entry`, `NODE_PATH`, :roll_eyes: )

File: package.json [hunk 2]
```diff
@@ -114,13 +114,15 @@
...
-    "i18n-calypso": "2.0.1",
+    "i18n-calypso": "file:packages/i18n",
     "ignore-loader": "0.1.2",
     "immutability-helper": "2.8.1",
     "immutable": "3.7.6",
     "imports-loader": "0.8.0",
     "inherits": "2.0.3",
+    "interpolate-components": "1.1.1",
     "is-my-json-valid": "2.19.0",
+    "jed": "1.0.2",
```
Comment (sirreal) [inline package.json]: Shouldn't `interpolate-components` and `jed` continue to be dependencies of the i18n package and not calypso proper? They are listed in the i18n package.json

Comment (ockham) [reply package.json]: Yeah, but that might need lerna in place. I didn't want to rebase on the lerna PR (yet).

**Comment by ockham** [comment]
Closing in favor of  #27489

**Pull request closed without merging**